stdlib/universe/pivot.go

```diff
@@ -667,14 +706,16 @@ func (t *sortedPivotTransformation) Finish(id execute.DatasetID, err error) {
 		return
 	}
 
-	var tbl flux.Table
-	gr := t.group
-	tbl, err = gr.doPivot(t.prevOutputKey, t.alloc)
-	if err != nil {
-		return
-	}
-	if err = t.d.Process(tbl); err != nil {
-		return
+	if t.group != nil {
+		var tbl flux.Table
+		gr := t.group
+		tbl, err = gr.doPivot(t.prevOutputKey, t.alloc)
+		if err != nil {
+			return
+		}
+		if err = t.d.Process(tbl); err != nil {
+			return
+		}
```

Comment:
This logic error was causing my pipeline test to fail.

Reply:
My understanding is that `t.group` should only be `nil` when there was never any data. Is there any other way for this to happen?

Reply:
Yeah, I think you may be right about that. Initially this was tied to another failure that I was saying, so this was not really written to prevent cases where there was never any data, but since you mentioned that this is good validation code for that particular case, I'll leave it it.